src/RevOpts.cc

```diff
@@ -49,226 +26,117 @@ void RevOpts::SetArgs( const SST::Params& params ) {
-    vstr.clear();
-  }
-  return true;
+  return InitPropertyMap( StartSymbols, startSym );
 }
 
+/// RevOpts: initialize the set of machine models
 bool RevOpts::InitMachineModels( const std::vector<std::string>& Machines ) {
-  std::vector<std::string> vstr;
-
-  // check to see if we expand into multiple cores
-  if( Machines.size() == 1 ) {
-    std::string s = Machines[0];
-    splitStr( s, ":", vstr );
-    if( vstr.size() != 2 )
-      return false;
-
-    if( vstr[0] == "CORES" ) {
-      // set all cores to the target machine model
-      for( uint32_t i = 0; i < numCores; i++ ) {
-        machine.at( i ) = vstr[1];
-      }
-      return true;
-    }
-  }
-
-  // parse individual core configs
-  for( uint32_t i = 0; i < Machines.size(); i++ ) {
-    std::string s = Machines[i];
-    splitStr( s, ":", vstr );
-    if( vstr.size() != 2 )
-      return false;
-
-    uint32_t Core = std::stoul( vstr[0], nullptr, 0 );
-    if( Core > numCores )
-      return false;
-
-    machine.at( Core ) = vstr[1];
-    vstr.clear();
-  }
-  return true;
+  auto [ret, retval] = InitPropertyMapCores( Machines, machine );
+  return ret ? retval : InitPropertyMap( Machines, machine );
 }
 
+/// RevOpts: initalize the set of instruction tables
 bool RevOpts::InitInstTables( const std::vector<std::string>& InstTables ) {
-  std::vector<std::string> vstr;
-  for( uint32_t i = 0; i < InstTables.size(); i++ ) {
-    std::string s = InstTables[i];
-    splitStr( s, ":", vstr );
-    if( vstr.size() != 2 )
-      return false;
-
-    uint32_t Core = std::stoul( vstr[0], nullptr, 0 );
-    if( Core > numCores )
-      return false;
+  return InitPropertyMap( InstTables, table );
+}
 
-    table.at( Core ) = vstr[1];
-    vstr.clear();
-  }
-  return true;
+/// RevOpts: initialize the prefetch depths
+bool RevOpts::InitPrefetchDepth( const std::vector<std::string>& Depths ) {
+  return InitPropertyMap( Depths, prefetchDepth );
 }
 
+/// RevOpts: initialize the memory latency cost tables
 bool RevOpts::InitMemCosts( const std::vector<std::string>& MemCosts ) {
-  std::vector<std::string> vstr;
-
-  for( uint32_t i = 0; i < MemCosts.size(); i++ ) {
-    std::string s = MemCosts[i];
+  for( auto& s : MemCosts ) {
+    std::vector<std::string> vstr;
     splitStr( s, ":", vstr );
     if( vstr.size() != 3 )
       return false;
-
-    uint32_t Core         = std::stoul( vstr[0], nullptr, 0 );
-    uint32_t Min          = std::stoul( vstr[1], nullptr, 0 );
-    uint32_t Max          = std::stoul( vstr[2], nullptr, 0 );
-    memCosts[Core].first  = Min;
-    memCosts[Core].second = Max;
-    if( ( Min == 0 ) || ( Max == 0 ) ) {
+    auto Core = std::stoull( vstr[0], nullptr, 0 );
+    auto Min  = decltype( memCosts[Core].first )( std::stoull( vstr[1], nullptr, 0 ) );
+    auto Max  = decltype( memCosts[Core].second )( std::stoull( vstr[2], nullptr, 0 ) );
+    if( Core >= numCores || !Min || !Max )
       return false;
-    }
```

Comment:
`InitMemCosts()` is different from the others because it has `{Min, Max}` pairs. But it does not need to be as general as the others, so it is fairly short code. The `decltype( memCosts[Core].first )` and `decltype( memCosts[Core].second )` are used to cast `stoull()` to the target `{Min, Max}` types, to prevent warnings in case they are smaller than `stoull()`.